### ISSUE TITLE
show more cluster info after installer finish

### ISSUE THREAD

**Issue by mfojtik** [openshift/installer#414]
As for now, when the installer finishes its job, it only prints how many resources were applied, but it does not tell me anything else...

Some suggestions for what we should print out after installer finish:

* A path to kubeconfig and how to use it (something users can copy&pasta)
* List of public hostnames for masters perhaps (including the `core@` username for login). Right now, I have to either go to EC2 console to figure out what it public hostname for my masters and bootstrap node is or parse the terraform vars...
* An API URL to use to login with the user I specified? (`oc login https://<hostname>:6443/`)
* The help (?) button in setup should probably also show the "hidden" environment variable you can set to not be asked the question on next installer run ;-)

Also it would be a good idea to warn that you haven't set the public SSH key variable, so you won't be able to SSH into your cluster, probably ask users if they are OK with that :-)

Also I noticed that the cluster is not fully up when the installer complete, which might not be great experience... If we don't want to poll the API server readiness in installer, we should probably tell users to wait for the cluster to fully provision ?

**Comment by wking** [comment]
There's a lot going on here.  Can you split it up into separate issues.  Waiting on API readiness, for example is linked to automatically removing the bootstrap resources, which @crawford is working on now.  But it's orthogonal to output after an `openshift-install cluster` call.  Focusing on your output suggestions:

> * A path to kubeconfig and how to use it (something users can copy&pasta)

I promoted this from the libvirt docs to the README [here][1].

> * List of public hostnames for masters perhaps (including the core@ username for login). Right now, I have to either go to EC2 console to figure out what it public hostname for my masters and bootstrap node is or parse the terraform vars...

`core@` is due to RHCOS, the installer doesn't really care.  If someone else brings their own host, the `core@` advice may not apply.  Most folks will be running this on RHCOS, which is why our libvirt docs can [talk about this][2].  But I'd be nervous about coding this image-specific information into the installer directly.

> * An API URL to use to login with the user I specified? (oc login https://<hostname>:6443/)

This is [in the README now][3] too.

> The help (?) button in setup should probably also show the "hidden" environment variable you can set to not be asked the question on next installer run ;-)

They're [not hidden][4] (linked from [here][4.5] ;).

> Also it would be a good idea to warn that you haven't set the public SSH key variable, so you won't be able to SSH into your cluster, probably ask users if they are OK with that :-)

You explicitly selected `<none>` and then expected to be able to SSH in?  I'm fine trusting users without second-guessing them.  If they get it wrong, they can always spin up a new cluster.  If we are really concerned about this, I'd be ok adjusting the string to be `<none> (will not be able to SSH into your cluster)` or some such.  But eventually we'll be able to [roll out new SSH keys via the machine-config-operator][5], and once we get there getting the SSH prompt wrong will be even less of an issue.

So overall, I think the recent #402 addresses most of your concerns, with the difference being that docs in this repo are not as discoverable as command-line output.  But promoting these to command-line output would mean that we'd either need to be much briefer, e.g. just dumping:

```
https://${OPENSHIFT_INSTALL_CLUSTER_NAME}-api.${OPENSHIFT_INSTALL_BASE_DOMAIN}:6443/console/
export KUBECONFIG="${DIR}/auth/kubeconfig"
```

with the variables expanded, but without the associated descriptions.  Or, if we added the associated descriptions, dumping a potentially confusing amount of docs into stdout.  Personally, I'm fine expecting users to read the docs for these sorts of things.  But if other folks have strong opinions about dumping some of this onto stdout, I won't put up too much of a fight ;).

[1]: https://github.com/openshift/installer/pull/402/files#diff-04c6e90faac2675aa89e2176d2eec7d8R42
[2]: https://github.com/openshift/installer/blob/v0.1.0/docs/dev/libvirt-howto.md#ssh-access
[3]: https://github.com/openshift/installer/pull/402/files#diff-04c6e90faac2675aa89e2176d2eec7d8R36
[4]: https://github.com/openshift/installer/blob/v0.1.0/docs/user/environment-variables.md
[4.5]: https://github.com/openshift/installer/blame/v0.1.0/README.md#L26
[5]: https://github.com/openshift/installer/blob/v0.1.0/pkg/asset/ignition/machine/node.go#L40

**Comment by wgordon17** [comment]
Would also be good to display console URL in the completion message

**Comment by crawford** [comment]
Yep, this seems to have been fixed in https://github.com/openshift/installer/pull/806.

/close
